int32 rid
float32 x
float32 y
float32 h
float32 priority
bool stopped
bool reached
